flink-ml-lib/src/test/java/org/apache/flink/ml/feature/VectorAssemblerTest.java

```diff
@@ -0,0 +1,168 @@
+                        .column("f2", DataTypes.DOUBLE())
+                        .column("f3", DataTypes.of(SparseVector.class))
+                        .build();
+        DataStream<Row> dataStream = env.fromCollection(INPUT_DATA);
+        inputDataTable =
+                tEnv.fromDataStream(dataStream, schema).as("id", "vec", "num", "sparseVec");
+    }
+
+    private void verifyPredictionResult(Table output, String outputCol) throws Exception {
+        DataStream<Row> dataStream = tEnv.toDataStream(output);
+        List<Row> results = IteratorUtils.toList(dataStream.executeAndCollect());
+        assertEquals(3, results.size());
+        for (Row result : results) {
+            if (result.getField(0) == (Object) 0) {
+                assertEquals(EXPECTED_DATA_1, result.getField(outputCol));
+            } else if (result.getField(0) == (Object) 1) {
+                assertEquals(EXPECTED_DATA_2, result.getField(outputCol));
+            } else {
+                assertNull(result.getField(outputCol));
+            }
+        }
+    }
+
+    @Test
+    public void testParam() {
+        VectorAssembler vectorAssembler = new VectorAssembler();
+        assertEquals(HasHandleInvalid.ERROR_INVALID, vectorAssembler.getHandleInvalid());
+        assertEquals("output", vectorAssembler.getOutputCol());
+        vectorAssembler
+                .setInputCols("vec", "num", "sparseVec")
+                .setOutputCol("assembledVec")
+                .setHandleInvalid(HasHandleInvalid.SKIP_INVALID);
+        assertArrayEquals(new String[] {"vec", "num", "sparseVec"}, vectorAssembler.getInputCols());
+        assertEquals(HasHandleInvalid.SKIP_INVALID, vectorAssembler.getHandleInvalid());
+        assertEquals("assembledVec", vectorAssembler.getOutputCol());
+    }
+
+    @Test
+    public void testTransform() throws Exception {
+        VectorAssembler vectorAssembler =
+                new VectorAssembler()
+                        .setInputCols("vec", "num", "sparseVec")
+                        .setOutputCol("assembledVec")
+                        .setHandleInvalid(HasHandleInvalid.SKIP_INVALID);
+        Table output = vectorAssembler.transform(inputDataTable)[0];
+        verifyPredictionResult(output, vectorAssembler.getOutputCol());
+    }
+
+    @Test
+    public void testHandleInvalidOptions() {
+        VectorAssembler vectorAssembler =
+                new VectorAssembler()
+                        .setInputCols("vec", "num", "sparseVec")
+                        .setOutputCol("assembledVec")
+                        .setHandleInvalid(HasHandleInvalid.ERROR_INVALID);
+        try {
+            Table outputTable = vectorAssembler.transform(inputDataTable)[0];
+            outputTable.execute().collect().next();
+            Assert.fail("Expected IllegalArgumentException");
+        } catch (Exception e) {
+            assertEquals(RuntimeException.class, ((Throwable) e).getClass());
```

Comment:
Can we check the detailed exception message? For example in `LogisticRegressionTest`
```java

            assertEquals(
                    "Multinomial classification is not supported yet. Supported options: [auto, binomial].",
                    e.getCause().getCause().getMessage());
```